### PR TITLE
Update CI and doc - fix for DataArrays

### LINKED ISSUE #224: Add note to doc about conda installation of esmpy
With esmpy 8.4 the conda recipe has changed. There a library file that now gets linked through an environment variable. When installing with conda, this environment variable is set upon _activation_ of the virtual env, and NOT when the installer runs. 

Thus, one can't do: 
```
conda activate myenv
conda install esmpy
python -c "import esmpy"
```
One must rather do:
```
conda activate myenv
conda install esmpy
conda deactivate
conda activate myenv
python -c "import esmpy"
```

This is issue : conda-forge/esmf-feedstock#91.

A note to the documentation should be added.

### LINKED ISSUE #225: Test against python 3.10 (3.11 ?)
Python 3.7 is slowly being deprecated and python 3.10 has been there for a while!  xESMF seems to often be used on machines with older python versions and by users not having all the permissions and power to update those, so we might not want to drop python 3.7 testing and support for the moment.

However, I think we should add a CI build for 3.10, and may be even for 3.11, to take some steps ahead?

### LINKED ISSUE #227: Dataset can't be DataArray
Hi! Thanks for this package, I use it regularly. 

I apparently have accidentally had testing for my functions that use xESMF turned off for awhile but I think this used to work so I wanted to note it.

When I run a `Regridder` command with an input DataArray instead of Dataset, I get an error like this:
```
    regridder = xe.Regridder(
../../miniconda3/envs/extract_model/lib/python3.9/site-packages/xesmf/frontend.py:773: in __init__
    grid_in, shape_in, input_dims = ds_to_ESMFgrid(
../../miniconda3/envs/extract_model/lib/python3.9/site-packages/xesmf/frontend.py:116: in ds_to_ESMFgrid
    lon, lat = _get_lon_lat(ds)
../../miniconda3/envs/extract_model/lib/python3.9/site-packages/xesmf/frontend.py:45: in _get_lon_lat
    if ('lat' in ds and 'lon' in ds) or ('lat' in ds.coords and 'lon' in ds.coords):

...

E           ValueError: The truth value of a Array is ambiguous. Use a.any() or a.all().
```

But I experimented and if I pause in the code there and convert "ds" to a Dataset, then it seems to run ok.

Should the xarray objects be Datasets for other reasons in the code, in which case a `TypeError` should probably be input if `ds_in` is a DataArray, or should that boolean statement be modified in case a DataArray is input?

Thanks!

### PR DESCRIPTION
**Pull request by aulemahal**
Fixes #224, fixes #225 , fixes #227.

- Accepts `DataArray` in `Regridder` and `SpatialAverager` for both  `ds_in` and `ds_out`, they are simply converted to `_temp_dataset` before doing anything. A trivial test was added.
- Expands the CI jobs to python 3.10 ~and 3.11~. A specific version of ESMpy was pinned for each version.
    - 3.11 was removed as numba (a dependency) is not yet compatible
    - I also removed the CI trigger on "push" (except for master). It was triggering all builds in duplicate when opening a pull request. Now, only pull requests and the master branch will have CI runs.
    - Push the upstream build to use python 3.10.
- Updated the installation page
   - Remove reference to specific python version
   - Remove warning for ESMF and mpi. I believe this was an issue before we put esmpy>=8.0 as a dependency.
   - Add new deps.
   - Remove note that pip is limited to 0.3
   - Add notes about `ESMpy` 8.4
- Updated the doc
   - Bumped the version number to 0.7.0.
   - Dynamical year and add "and the xESMF development team" to the copyright.
   - Mock import for esmpy.

### CODE REVIEW

**Comment by aulemahal** [comment]
@raphaeldussin Do you know what are those "build (3.7) _Expected_" entries in the checks? I think they never get updated since I changed the build names, but I can't find where these entries are set to update them...

Maybe @Zeitsperre knows about this.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 name: CI
 
 on:
-  push:
-    branches:
-      - '*'
```
Comment (Zeitsperre) [inline .github/workflows/ci.yaml:6]: I would just list the main branch here to ensure that master isn't borked when it is updated.

File: .github/workflows/linting.yaml [hunk 2]
```diff
@@ -1,8 +1,6 @@
 name: linting
 
 on:
-  push:
-    branches: '*'
```
Comment (Zeitsperre) [inline .github/workflows/linting.yaml:5]: idem

**Comment by Zeitsperre** [comment]
> @raphaeldussin Do you know what are those "build (3.7) _Expected_" entries in the checks? I think they never get updated since I changed the build names, but I can't find where these entries are set to update them...
> 
> Maybe @Zeitsperre knows about this.

In the branch settings, at some point, someone with maintainer access marked some CI actions as necessary in order to merge to the main development branch. This need to be updated to use the existing CI jobs instead.

**Comment by aulemahal** [comment]
Ok thanks @Zeitsperre ! I don't have access to that I believe.

**Comment by huard** [comment]
Removed expected builds from checks

**Review by raphaeldussin — APPROVED** [review]
LGTM! thanks @aulemahal

**Review by huard — APPROVED** [review]
Well done !

**Pull request merged**